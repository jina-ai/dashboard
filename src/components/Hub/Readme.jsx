import React from "react";
import { Card, CardHeader, CardBody } from "shards-react";

export default function ReadMe({ readme, documentation }) {
  return (
    <Card className="readme-container mb-4">
      <CardHeader className="border-bottom d-flex flex-row">
        <h6 className="m-0 d-inline-block">README.MD</h6>
        <div className="flex-fill d-inline-block" />
        <a
          href={documentation}
          target="_blank"
          rel="noopener noreferrer"
          className="cursor-pointer text-primary"
        >
          <i className="fab fa-github"></i> View on GitHub
        </a>
      </CardHeader>
      <CardBody>
        {readme ? (
          <div
            className="markup"
            dangerouslySetInnerHTML={{ __html: readme }}
          />
        ) : (
          <h2 className="text-muted text-center py-4">No Readme Found</h2>
        )}
      </CardBody>
    </Card>
  );
}
