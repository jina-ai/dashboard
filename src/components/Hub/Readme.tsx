import React, { useState, useEffect } from "react"
import { Card, CardHeader, CardBody } from "shards-react"
import { getDocumentationHTML } from "../../services/hubApi"
import SpinningLoader from "../Common/SpinningLoader"

type Props = {
  documentation: string
}

export default function ReadMe({ documentation }: Props) {
  let [readme, setReadme] = useState("")
  useEffect(() => {
    ;(async () => {
      const readmeHTML = await getDocumentationHTML(documentation)
      setReadme(readmeHTML)
    })()
  }, [documentation])
  return (
    <Card className="readme-container mb-4">
      <CardHeader className="border-bottom d-flex flex-row">
        <h6 className="m-0 d-inline-block">README.md</h6>
        <div className="flex-fill d-inline-block" />
        <a
          href={documentation}
          target="_blank"
          rel="noopener noreferrer"
          className="cursor-pointer text-primary"
        >
          <i className="fab fa-github"></i> View on GitHub
        </a>
      </CardHeader>
      <CardBody>
        {readme ? (
          <div
            className="markup"
            dangerouslySetInnerHTML={{ __html: readme }}
          />
        ) : (
          <SpinningLoader />
        )}
      </CardBody>
    </Card>
  )
}
