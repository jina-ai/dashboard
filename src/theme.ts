const PERSIAN_GREEN = "#009999"
const GOLDEN_TANOI = "#FFCC66"
const BITTER_SWEET = "#ff6666"
const SCOOTER_GREEN = "#32C8CD"
const WHITE = "#FFFFFF"
const DADGREY = "#DAD7FE"
const SKY_DIVING_IN_THE_BLUE = "#007AFF"
const QUICK_SILVER = "#A6A6A6"
const HONEYBEE = "#FFCC66"
const BLACK = "#000000"

const theme = {
  palette: {
    background: WHITE,
    headerTextColor: BLACK,
    primary: PERSIAN_GREEN,
    secondary: GOLDEN_TANOI,
    warning: GOLDEN_TANOI,
    error: BITTER_SWEET,
    success: SCOOTER_GREEN,
    info: PERSIAN_GREEN,
    critical: "#ff4540",
    debug: "#7E7278",
    tagBackground: DADGREY,
    tagPlatformBackground: HONEYBEE,
    mutedText: QUICK_SILVER,
    highlight: SKY_DIVING_IN_THE_BLUE,
  },
}

export { theme }
