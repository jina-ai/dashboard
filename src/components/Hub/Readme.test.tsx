import React from "react"
import renderer from "react-test-renderer"
import { ThemeProvider } from "@emotion/react"
import { theme } from "../../theme"
import Readme from "./Readme"

test("Readme", () => {
  const component = renderer.create(
    <ThemeProvider theme={theme}>
      <Readme
        documentation={
          "https://raw.githubusercontent.com/jina-ai/jina-hub/master/encoders/image/BigTransferEncoder/README.md"
        }
      />
    </ThemeProvider>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
