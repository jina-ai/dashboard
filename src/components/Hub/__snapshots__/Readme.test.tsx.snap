// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Readme 1`] = `
<div
  className="readme-container mb-4 card"
>
  <div
    className="border-bottom d-flex flex-row card-header"
  >
    <h6
      className="m-0 d-inline-block"
    >
      README.md
    </h6>
    <div
      className="flex-fill d-inline-block"
    />
    <a
      className="cursor-pointer text-primary"
      href="https://raw.githubusercontent.com/jina-ai/jina-hub/master/encoders/image/BigTransferEncoder/README.md"
      rel="noopener noreferrer"
      target="_blank"
    >
      <i
        className="fab fa-github"
      />
       View on GitHub
    </a>
  </div>
  <div
    className="card-body"
  >
    <div
      className="css-18vjixk"
    >
      <div
        className="css-11ue89k"
      >
        <div
          className="css-jn4y13"
        />
      </div>
    </div>
  </div>
</div>
`;
